app/tx_test.go

```diff
@@ -344,3 +345,174 @@ func TestClaimTx(t *testing.T) {
 		}
 	}
 }
+
+func TestClaimTxChallenge(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping in short mode")
+	}
+	pocketTypes.GetEvidenceMap().Clear()
+	genBz, keys, _, _ := fiveValidatorsOneAppGenesis()
+	challenges := NewValidChallengeProof(t, keys, 5)
+	for _, c := range challenges {
+		err := c.Handle()
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
+	}
+	_, _, cleanup := NewInMemoryTendermintNode(t, genBz)
+	_, stopCli, evtChan := subscribeTo(t, tmTypes.EventTx)
+	select {
+	case res := <-evtChan:
+		if res.Events["message.action"][0] != pocketTypes.EventTypeClaim {
+			t.Fatal("claim message was not received first")
+		}
+		_, stopCli, evtChan = subscribeTo(t, tmTypes.EventTx)
+		select {
+		case res := <-evtChan:
+			if res.Events["message.action"][0] != pocketTypes.EventTypeProof {
+				t.Fatal("proof message was not received afterward")
+			}
+			cleanup()
+			stopCli()
+		}
+	}
+}
+
+func NewValidChallengeProof(t *testing.T, privateKeys []crypto.PrivateKey, numOfChallenges int) (challenge []pocketTypes.ChallengeProofInvalidData) {
```

Comment:
Do we need a test to try the functionality of the "unsorted JSON" sorting?

Reply:
Will add